.github/workflows/check_notebooks.yml

```diff
@@ -0,0 +1,41 @@
+name: Check notebook's integrity
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run
+    # by the push to the branch.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10']
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Check whether tutorials were changed
+      id: changed
+      continue-on-error: true # Non-empty diff implies failure.
+      run: |
+        git diff HEAD~1 HEAD \
+        --no-patch \
+        --exit-code \
+        $GITHUB_WORKSPACE/examples/tutorials/*ipynb
+    - name: Install package and development dependencies
+      if: steps.changed.outcome == 'failure'
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e ${GITHUB_WORKSPACE}[style,test,wheel]
+    - name: Make sure tutorials run
+      if: steps.changed.outcome == 'failure'
+      run: |
+        (cd .. && pytest --nbmake $GITHUB_WORKSPACE/examples/tutorials/*.ipynb)
```

Comment:
```suggestion
    - name: Install package and development dependencies
      run: |
        python -m pip install --upgrade pip
        pip install -e ${GITHUB_WORKSPACE}[style,test,wheel]
    - name: Make sure tutorials run
      run: |
        (cd .. && pytest --nbmake $GITHUB_WORKSPACE/examples/tutorials/*.ipynb)
```

Reply:
Closes #472 as it was unnecessary in the end: Notebooks should always be tested.